### PR TITLE
DPC-1645: Remove eval from implementer app

### PR DESCRIPTION
**Pull request by Sun-Mountain**
### Fixes [DPC-1645](https://jira.cms.gov/browse/DPC-1645)

Remove eval from the impl app.

### Proposed Changes

Because we want to simplify code and leave some of the guess work out. By removing the `eval()` ruby command and replacing it with the `JSON.parse` function, we are being clearer with what is by the frontend (a JSON object).

### Change Details

As the client token is the only one returned as a string rather than an object. For consistency, I recommend that the api return the client token as a JSON object and the rescue can return a simple error message. I created the ticket [DPC-1815](https://jira.cms.gov/browse/DPC-1815) to make the api calls more consistent.

### Security Implications

Eliminating `eval()` closes the door for taking over the implementer portal via a malicious response from the API.

- [ ] new software dependencies

- [ ] security controls or supporting software altered

- [ ] new data stored or transmitted

- [x] [security checklist](https://confluence.cms.gov/pages/viewpage.action?spaceKey=DAPC&title=2021-09-02%3A+DPC+4.2+Security+Checklist) is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

- [x] no PHI/PII is affected by this change

### Acceptance Validation

Screen Shot 2021-09-07 at 3 35 45 PM

### CODE REVIEW

File: dpc-impl/app/models/user.rb [hunk 1]
```diff
@@ -69,7 +68,7 @@ def provider_orgs
     api_client = ApiClient.new
     api_client.get_provider_orgs(self.implementer_id)
     if api_client.response_successful?
-      api_client.response_body.sort_by{|e| e[:created_at]}.reverse
+      api_client.response_body.sort_by{|e| e["created_at"]}.reverse
```
Comment (dhgreene) [inline dpc-impl/app/models/user.rb:71]: Shall we use single quotes as above?
```suggestion
      api_client.response_body.sort_by{|e| e['created_at']}.reverse
```

File: dpc-impl/app/models/user.rb [hunk 2]
```diff
@@ -43,11 +43,10 @@ def check_impl
 
   def create_api_imp
     api_request = api_service.create_implementer(implementer)
-
     api_response = api_request.response_body
 
     if api_request.response_successful?
-      self.implementer_id = api_response[:id]
+      self.implementer_id = api_response["id"]
```
Comment (dhgreene) [inline dpc-impl/app/models/user.rb:49]:
```suggestion
      self.implementer_id = api_response['id']
```

File: dpc-impl/app/services/api_client.rb [hunk 3]
```diff
@@ -113,9 +113,12 @@ def parsed_response(response)
     begin
       !!JSON.parse(response.body)
     rescue
+      # Returns client token
+      # TODO: have client token return as JSON object for consistency
```
Comment (dhgreene) [inline dpc-impl/app/services/api_client.rb:117]: Do we have a ticket to follow up on this TODO promise?

**Review by dhgreene — COMMENTED** [review]
The PR description needs a little love. In particular: 
1. [Here's the link](https://confluence.cms.gov/display/DAPC/2021-09-02%3A+DPC+4.2+Security+Checklist) to the security checklist for this ticket
2. Note that this work is a security improvement; eliminating`eval()` closes the door for taking over the implementer portal via a malicious response from the API

File: dpc-impl/app/services/api_client.rb (hunk 3 above)
Comment (Sun-Mountain) [reply dpc-impl/app/services/api_client.rb:117]: I will create one and link it in the PR

Comment (Sun-Mountain) [reply dpc-impl/app/services/api_client.rb:117]: Created [DPC-1815](https://jira.cms.gov/browse/DPC-1815)

**Review by dhgreene — APPROVED** [review]
😌

**Pull request merged**